src/Storages/MergeTree/MergeTreePartsMover.cpp

```diff
@@ -240,14 +243,22 @@ MergeTreeMutableDataPartPtr MergeTreePartsMover::clonePart(const MergeTreeMoveEn
     {
         cloned_part_storage = part->makeCloneOnDisk(disk, MergeTreeData::MOVING_DIR_NAME);
     }
+    String data_part_directory = cloned_part_storage->getFullPath();
+
+    TemporaryClonedPart cloned_part;
+    cloned_part.temporary_directory_lock = data->getTemporaryPartDirectoryHolder(data_part_directory);
```

Comment:
> Does/should data_part_directory include moving/?

Yes, since we lock part in `moving` directory before it's renamed.

```cpp
cloned_part_storage = part->makeCloneOnDisk(disk, MergeTreeData::MOVING_DIR_NAME);
```

Reply:
Holder should accept only base name of the part https://github.com/ClickHouse/ClickHouse/blob/5d0522cfa6045ccecc77f79dd12b043260e1545b/src/Storages/MergeTree/MergeTreeData.cpp#L2095

Fixed